Core/ModuleInstaller.cs

```diff
@@ -989,20 +964,23 @@ public void Upgrade(IEnumerable<string> identifiers, IDownloader netAsyncDownloa
         /// Will *re-install* or *downgrade* (with a warning) as well as upgrade.
         /// Throws ModuleNotFoundKraken if a module is not installed.
         /// </summary>
-        public void Upgrade(IEnumerable<CkanModule> modules, IDownloader netAsyncDownloader, ref HashSet<string> possibleConfigOnlyDirs, RegistryManager registry_manager, bool enforceConsistency = true, bool resolveRelationships = false)
+        public void Upgrade(IEnumerable<CkanModule> modules, IDownloader netAsyncDownloader, ref HashSet<string> possibleConfigOnlyDirs, RegistryManager registry_manager, bool enforceConsistency = true, bool resolveRelationships = false, bool ConfirmPrompt = true)
         {
             modules = modules.Memoize();
-            User.RaiseMessage("About to upgrade...\r\n");
+            User.RaiseMessage("About to upgrade:\r\n");
 
             if (resolveRelationships)
             {
-                var resolver = new RelationshipResolver(modules, null, RelationshipResolver.DependsOnlyOpts(), registry_manager.registry, ksp.VersionCriteria());
+                var resolver = new RelationshipResolver(
+                    modules,
+                    modules.Select(m => registry_manager.registry.InstalledModule(m.identifier)?.Module).Where(m => m != null),
```

Comment:
Hmm, I'm not able to confirm this (which makes sense to me, since there's no reason `.Select()` shouldn't be allowed to return `null`). I tried forcing it (since I think this could never actually contain a `null` for `ckan upgrade --all`), and it made it past this part just fine:

```
                    modules.Select(m => (CkanModule)null).Where(m => m != null),
```

```
$ _build/ckan.exe upgrade --all
About to upgrade:

 * Upgrade: Kopernicus Bleeding Edge Beta - DEV RELEASE UBEE_1101_44 to UBEE_1101_46 (cached)

Continue? [Y/n] 

Removing "Kopernicus-BE UBEE_1101_44"
Installing "Kopernicus-BE UBEE_1101_46"
Updating registryThe following inconsistencies were found:
* Kopernicus-BE UBEE_1101_46 missing dependency ModularFlightIntegrator 1.2.7.0 or later
```

I suspect your changes were hitting a different issue.

Reply:
Interesting. Can't reproduce myself any more. Maybe it was `.Where()` or `.Select()` where I encountered this and I confused them?
So yeah, ignore that, sorry.